### PR TITLE
GET | /kindergarten

### PR DESCRIPTION
**Pull request by alaa-sahloub**
Relates #12

### Create getKindergartenQuery.js file which Contain:

- Import the database connection.
- create getKindergartenQuery inside it declare sql as query:
``` SELECT *, (SELECT AVG(comments.rating) AS rating_average FROM comments WHERE comments.kindergarten_id = kindergartens.id), (SELECT COUNT(comments) AS rating_count FROM comments WHERE comments.kindergarten_id = kindergartens.id) FROM kindergartens``` 
- sql connectin with sql query.
- export getKindergartenQuery.

### Create getKindarten.js which contain:

- Import the getKindergartenQuery form queries.
- Create the getKindergarten function using async-await.
- In the try invoke the getKindergartenQuery() and return statusCode 200 and the data.
- OR catch the error.
- Export the getKindergarten.

### In the kindergartens.js inside the router folder:

-  import the getKinergarten from the controllers.
-  create the router to /kindergarten endpoint with getKinergarten function.
- export the router.

### CODE REVIEW

File: server/controllers/index.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// const {clientError, serverError} = require('./errorHandle');
+// const {verifyUser, verifyAdmin} = require('./auth');
```
Comment (Mu7ammadAbed) [inline server/controllers/index.js:2]:
```suggestion
const {clientError, serverError} = require('./errorHandle');
```

File: server/controllers/kindergartens/getKindergarten.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// const { default: axios } = require('axios');
```
Comment (Mu7ammadAbed) [inline server/controllers/kindergartens/getKindergarten.js:1]: There's no need to use Axios here

File: test/getKindergarten.test.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const request = require('supertest');
```
Comment (Mu7ammadAbed) [inline test/getKindergarten.test.js:1]: Move the content of this file to `index.test.js`

File: package.json [hunk 4]
```diff
@@ -25,6 +25,7 @@
   },
   "homepage": "https://github.com/GSG-G9/Rawdaty#readme",
   "dependencies": {
+    "axios": "^0.21.1",
```
Comment (Mu7ammadAbed) [inline package.json:28]: uninstall this

File: test/getKindergarten.test.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+      .get('/api/v1/kindergarten')
+      .expect(200)
+      .expect('Content-Type', /json/)
+      .end((err, res) => {
+        if (err) return done(err);
+        const { data } = res.body;
+        expect(data).toHaveLength(11);
+        return done();
+      }));
+});
```
Comment (Mu7ammadAbed) [inline test/getKindergarten.test.js:21]: Change this test to `async/await`

```suggestion
describe('Get all kindergartens', () => {
  test('Route /kindergarten status 200, json header', async () => {
    expect.assertions(1);
    const res = await request(app)
      .get('/api/v1/kindergarten')
      .expect(200)
      .expect('Content-Type', /json/);
    const { data } = res.body;
    expect(data).toHaveLength(11);
  });
});
```

File: server/database/queries/kindergarten/getKindergartenQuery.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+const connection = require('../../data/connection');
+
+const getKindergartenQuery = () => {
+  const sql = {
+    text:
+      'SELECT *, (SELECT AVG(comments.rating) AS rating_average FROM comments WHERE comments.kindergarten_id = kindergartens.id), (SELECT COUNT(comments) AS rating_count FROM comments WHERE comments.kindergarten_id = kindergartens.id) FROM kindergartens',
```
Comment (Mu7ammadAbed) [inline server/database/queries/kindergarten/getKindergartenQuery.js:6]:
```suggestion
      'SELECT *, (SELECT AVG(comments.rating) AS rating_average FROM comments WHERE comments.kindergarten_id = kindergartens.id), (SELECT COUNT(comments) AS rating_count FROM comments WHERE comments.kindergarten_id = kindergartens.id) FROM kindergartens WHERE kindergartens.request_status = 'approved' AND kindergartens.is_enable = 'true'',
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Review by Osama-you — APPROVED** [review]

**Pull request merged**